### PR TITLE
Implement Default for RwLock and Mutex

### PR DESCRIPTION
**Pull request by ghost**
Hello! I have implemented `Default` for `Mutex` and `RwLock`, the std versions support it and I think it's very useful.

### CODE REVIEW

File: src/mutex.rs [hunk 1]
```diff
@@ -210,9 +218,17 @@ pub struct Mutex<T: ?Sized> {
     inner: sync::Arc<Inner<T>>,
 }
 
+impl<T: Default + ?Sized> Default for Mutex<T> {
+    fn default() -> Self {
+        Self::new(Default::default())
```
Comment (asomers) [inline src/mutex.rs]: I think it would be more idiomatic and readable to call this `T::default`.

**Review by asomers — CHANGES_REQUESTED** [review]
Yes, I think Default would be a good thing to implement.  But I have three comments:
1) There's a lot of line noise here.  Did you run rustfmt?  If you're going to do that, you must do it in a separate commit.  Mixing content changes with format changes is generally poor for maintainability.
2) This needs a CHANGELOG entry
3) It needs tests, do
4) Rather than manually implement `Default`, I think it could be derived if you also derive `Default` for `MutexData` and `RwLockData`.  That would also allow removing a few lines from `Mutex::new` and `RwLock::new`.

**Comment by ghost** [comment]
Hello @asomers I have fixed your comments

1. > There's a lot of line noise here. Did you run rustfmt?

    Yes, sorry, I have my editor configured to do that when saving and I forgot to check, I have left the format untouched this time

2. > This needs a CHANGELOG entry

    Problem is that the changes there need to point to a tagged version right?
3. > It needs tests, do

    Done!
4. > Rather than manually implement Default, I think it could be derived if you also derive Default ...

    Totally right, I have removed my previous code and just derived in the appropriate places

**Comment by asomers** [comment]
>     2. > This needs a CHANGELOG entry
>        
>        
>        Problem is that the changes there need to point to a tagged version right?

Just add a new section at the top, like this:
```markdown
## [Unreleased] - ReleaseDate
### Added
### Changed
### Fixed
### Removed
```

**Comment by ghost** [comment]
Done :heavy_check_mark:

**Comment by asomers** [comment]
Thanks for the CHANGELOG entry.  However, when I said "ReleaseDate", I meant it literally.  Just put a literal "ReleaseDate" in the CHANGELOG, and whenever I make a release I'll set it appropriately.

**Comment by ghost** [comment]
Oh I see, now it makes a lot more sense :smile:

**Review by asomers — APPROVED** [review]

**Pull request merged**